### PR TITLE
[16.01] Fix virtual env handling broken with #1847.

### PR DESCRIPTION
**Pull request by jmchilton**
Thanks for the bug report @natefoo.

### CODE REVIEW

**Comment by nsoranzo** [comment]
Should we add tests to Travis to check that these commands works, like we do for `run.sh` ?

**Comment by jmchilton** [comment]
I would say yes, but the existing travis check for run.sh is kind of noisy right? Does anyone know why it fails semi-frequently?

**Comment by dannon** [comment]
Kinda nitpicky, but now it prints twice:

```
[yoplait@anvil:~/galaxy] sh run_reports.sh                                                                                                                                                                                     29c78e9 ✭
Activating virtualenv at .venv
Activating virtualenv at .venv
```

**Comment by nsoranzo** [comment]
@jmchilton Is it? I haven't noticed many failures, but maybe somebody re-run the builds superfast.

**Comment by jmchilton** [comment]
I re-ran like 3 or 4 times - but admittedly it was all within a couple days about a week ago. Maybe it was some transient wheel or Travis problem that has been resolved.

**Comment by jmchilton** [comment]
@dannon good point, I've removed the print statements added with this PR.

**Comment by nsoranzo** [comment]
@dannon Actually also `run.sh` prints it twice, one for `scripts/common_startup.sh` and one for itself.

**Comment by natefoo** [comment]
It always printed it twice.

**Comment by dannon** [comment]
Sounds like we need to fix that in one more spot, then :)

Anyway, +1 for this change to 16.01.

**Comment by nsoranzo** [comment]
I think the double print is useful when debugging problems with virtual envs to check that `scripts/common_startup.sh` is sourcing the same env of `run.sh` . See also my next comment.

File: run_reports.sh [hunk 1]
```diff
@@ -11,6 +11,13 @@
 
 cd `dirname $0`
 
+: ${GALAXY_VIRTUAL_ENV:=.venv}
+
+if [ -d "$GALAXY_VIRTUAL_ENV" ];
+then
+    . "$GALAXY_VIRTUAL_ENV/bin/activate"
+fi
+
```
Comment (nsoranzo) [inline run_reports.sh:22]: This code must go AFTER `./scripts/common_startup.sh --skip-samples`, which will create the `.venv` directory if it does not exist.

File: run_tool_shed.sh [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 cd `dirname $0`
 
+: ${GALAXY_VIRTUAL_ENV:=.venv}
+
+if [ -d "$GALAXY_VIRTUAL_ENV" ];
+then
+    . "$GALAXY_VIRTUAL_ENV/bin/activate"
+fi
+
```
Comment (nsoranzo) [inline run_tool_shed.sh:13]: This code must go AFTER `./scripts/common_startup.sh`, which will create the `.venv` directory if it does not exist.

**Comment by jmchilton** [comment]
I've modified the order as requested by @nsoranzo.

**Comment by natefoo** [comment]
Merging this but we should probably switch to sourcing, further development in #1899

**Pull request merged**